Fixed Spark UI Tutorial - Addressing Previous Feedback

AmitKW11 opened the pull request:
Hi Noam,

fixed version.

-Added images for every tab with explanations about them. I used images from the SPARK UI official guide and not as URLS to be on the safe side.
-dropped unnecessary and non-comprehensive parts
-fixed corrupted cells and changed many of them in ways I found easier to learn.
I think the notebook is much more ''educating'' now.

It's been super challenging for me since it's not an easy subject at all to grasp, and also I had to face many technical incapacities that I managed to acquire and DEAL with them (like working with git!!!)
I'm grateful for this.

I apologize for the long replies, I am in milium (5 weeks. apparently we didn't win yet.........)
But I will do my best to work as fast as possible on whatever comments you have.

AmitKW11 committed:
fixed issues

cnoam commented:
very nice!
One last thing:  
1. Almost all the notebook is just reading, and the only code is at the end. 
You write "another example" so there should be one before it.
2. Once I run the cell (with the sql), I dont have feedback such as showing me some of the UI screenshots for this specific run.  Can you add a couple images with (short) explanations?

AmitKW11 commented:
1. Making sure I understand: are you asking to add more coding examples and screenshots for specific runs? i understood you correctly?
And if so, any tab in particular or all around? 

2. Got it, no problem.

cnoam commented:
add one small code example and explain on one tab (e.g. the DAG) how it relates to the commands in the code. 
This should give the reader the connection between the code and the UI.

AmitKW11 committed:
saved changes

cnoam commented:
The link is not working: I see this text, without an image. You should try to run it locally - in a clean folder - before pushing to git. it will show you if you forgot to push some contents. 
"
What to Look for in the Jobs Tab: After running this code, go to the Jobs tab in the Spark UI where you'll see: [SCREENSHOT 1: Jobs Overview] Take a screenshot of the main Jobs tab showing all 4 jobs with their descriptions, durations, and stages. This clearly shows how each action in your code creates a separate job in Spark.
"

noamkerentech committed:
fixed version

AmitKW11 commented:
Yes, the previous push was not a final version. It was pushed so I could pull it from my girlfriend's computer. 
For some reason, when running the code on my computer, I encountered compatibility issues with pyspark and python. Something about the JVM and Python processes in pyspark having connection issues .

 -I  had no idea it is visible to you :)

Anyway - the final fix has been pushed :

1. Added code exmaples to DAG section and also Job section
2. Added screenshots from the specific runs + relatively short explanations
3. Added  screenshots from last code example + explanations.
